### PR TITLE
MM-24481: Use atomics for load/store of client configs

### PR DESCRIPTION
**Pull request by agnivade**
There are 3 client config related variables which are set in regenerateClientConfig
but can be read from public methods in the App struct.

We use the same approach as the license fields in the App struct and use atomic.Value
for modifying them.

### Ticket link

https://mattermost.atlassian.net/browse/MM-24481

### CODE REVIEW

File: app/config.go [hunk 1]
```diff
@@ -319,9 +319,9 @@ func (a *App) regenerateClientConfig() {
 	}
 
 	clientConfigJSON, _ := json.Marshal(clientConfig)
-	a.Srv().clientConfig = clientConfig
-	a.Srv().limitedClientConfig = limitedClientConfig
-	a.Srv().clientConfigHash = fmt.Sprintf("%x", md5.Sum(clientConfigJSON))
+	a.Srv().clientConfig.Store(clientConfig)
+	a.Srv().limitedClientConfig.Store(limitedClientConfig)
+	a.Srv().clientConfigHash.Store(fmt.Sprintf("%x", md5.Sum(clientConfigJSON)))
```
Comment (streamer45) [inline app/config.go:324]: Just want to confirm that we are making sure that this is always going to execute before any call that loads the value (e.g. `ClientConfig()`).

Comment (agnivade) [reply app/config.go:324]: Yes, this gets called during server startup from `RunOldAppInitialization` using our wonderful FakeApp adapter.

**Comment by agnivade** [comment]
This will be hard for Lindy to test. I will request Claudio to do some peer testing and confirm that it's fixed, since it was caught in the load test originally.

**Comment by lindy65** [comment]
Thanks @agnivade - I've added the QA: Review Done label and will pass this through from QA's perspective 👍 

Will @streamer45 be peer testing here on the PR as well as post-merge on the resolved ticket?

**Comment by streamer45** [comment]
> Will @streamer45 be peer testing here on the PR as well as post-merge on the resolved ticket?

Yes. I can take care of this.

**Pull request merged**